app/models/sample.rb

```diff
@@ -209,12 +210,34 @@ def check_status
     kickoff_pipeline
   end
 
+  def archive_old_pipeline_runs
+    old_pipeline_runs = pipeline_runs.order('id desc').offset(1)
+    old_pipeline_runs.each do |pr|
+      # Write pipeline_run data to file
+      file = Tempfile.new
+      file.write(pr.to_json(include: :pipeline_run_stages))
+      file.close
+      # Copy file to S3
+      pr_s3_file_name = "pipeline_run_#{pr.id}.json"
+      command = "aws s3 cp #{file.path} #{pr.archive_s3_path}/#{pr_s3_file_name}"
+      _stdout, _stderr, status = Open3.capture3(command)
+      # Delete pipeline_run
```

Comment:
Hm I think we should keep the database records but remove the taxon_counts, taxon_byte_ranges. because if we delete all the records, we won't know what's going on and we can't link back to the cloud watch logs.

Reply:
Oh I see